sk0-back/src/users/controllers/users.controller.ts

```diff
@@ -112,14 +117,20 @@ export class UsersController {
     description: "The found pathed by id",
     type: UpdateUserDto
   })
+  @UseGuards(AuthGuard("jwt"))
   async updateUserById(
     @Param("id") id: string,
-    @Body() userData: Partial<CreateUserDto>
+    @Body() userData: Partial<CreateUserDto>,
+    @Req() req: Request
   ): Promise<UpdateUserDto> {
     const userId = parseInt(id, 10);
+    const user = req.user as User;
```

Comment:
I think we should rather use the extension of the `Request` type instead of casting, since user as a filed only exists on the field with a type if we are adding it, until that we have to assume it exists with a casting.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/07c771f1e113e2ebcd6a88778555883f22166ab3/types/express-serve-static-core/index.d.ts#L6-L15

Reply:
that makes sense.